h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -739,14 +746,24 @@ else if (ParseUUID.isUUID(str))
     return resSetup;
   }
 
-  private static String[] getFirstLines(byte[] bits) {
+  private static String[] getFirstLines(byte[] bits, boolean singleQuotes) {
```

Comment:
It seems like you are introducing a new functionality here (comments), this should be handled by a separate PR in my opinion.

Also it seems like once you set the comment to true you never unset it. Do we need this to fix PUBDEV-5394?

Reply:
In my opinion we need this.

To fix 5394, the counting of quotes is required. And thiese adjustments are required for the tests to work. In this particular case, there is a `test_parse_mix` that uses both types of quotes (as apostrophe). There are other datasets like this. In the GuessSetup phase, there is no info about quotes, only the one coming from the user. the singleQuotes boolean is not a new functionality.

Reply:
yep, my concern was the comments (handling of '#'), it doesn't seem like the comment flag is ever reset back to false

Reply:
@michalkurka That should be indeed ! With each new line ! Added and pushed !